src/main/java/com/thealgorithms/divideandconquer/MaximumSubarrayAlgorithm.java

```diff
@@ -0,0 +1,77 @@
+package com.thealgorithms.divideandconquer;
+
+public class MaximumSubarrayAlgorithm {
+
+    /**
+     * This class implements Kadane's algorithm to find the maximum subarray of a given array of integers.
+     * Kadane's algorithm runs in O(n) time complexity.
+     *
+     * More information about the Maximum Subarray problem can be found at:
+     * https://en.wikipedia.org/wiki/Maximum_subarray_problem
+     */
+
+
+    /**
+     * Finds the maximum subarray crossing the middle index.
+     *
+     * @param A the array to search for the maximum subarray
+     * @param low the lower index of the subarray to search
+     * @param mid the middle index of the subarray to search
+     * @param high the upper index of the subarray to search
+     * @return an array containing the indices and sum of the maximum subarray crossing the middle index
+     */
+    public static int[] findMaxCrossingSubarray(int[] A, int low, int mid, int high) {
+        int leftSum = Integer.MIN_VALUE;
+        int sum = 0;
+        int maxLeft = 0;
+
+        for (int i = mid; i >= low; i--) {
+            sum += A[i];
+            if (sum > leftSum) {
+                leftSum = sum;
+                maxLeft = i;
+            }
+        }
+
+        int rightSum = Integer.MIN_VALUE;
+        sum = 0;
+        int maxRight = 0;
+
+        for (int j = mid + 1; j <= high; j++) {
+            sum += A[j];
+            if (sum > rightSum) {
+                rightSum = sum;
+                maxRight = j;
+            }
+        }
+
+        return new int[]{maxLeft, maxRight, leftSum + rightSum};
+    }
+
+    /**
+     * Finds the maximum subarray within a given range of indices.
+     *
+     * @param A the array to search for the maximum subarray
+     * @param low the lower index of the subarray to search
+     * @param high the upper index of the subarray to search
+     * @return an array containing the indices and sum of the maximum subarray within the given range
+     */
+    public static int[] findMaximumSubarray(int[] A, int low, int high) {
```

Comment:
Please add a public function `int[] findMaximumSubarray(int[] array)` that only accepts array and returns only starting and ending indexes of the maximum subarray. And make these functions private, as those are implementation and not contract. They can be tested through the public function.